Gordon360/Services/RecIM/ActivityService.cs

```diff
@@ -0,0 +1,220 @@
+            _seriesService = seriesService;
+        }
+
+        public IEnumerable<LookupViewModel> GetActivityLookup(string type)
+        {
+            switch (type)
+            {
+                case "status":
+                    return _context.ActivityStatus
+                                .Select(s => new LookupViewModel
+                                {
+                                    ID = s.ID,
+                                    Description = s.Description
+                                })
+                                .AsEnumerable();
+                case "activity":
+                    return _context.ActivityType
+                                .Select(a => new LookupViewModel
+                                {
+                                    ID = a.ID,
+                                    Description = a.Description
+                                })
+                                .AsEnumerable();
+                default:
+                    return null;
+
+            }
+        }
+        public IEnumerable<ActivityExtendedViewModel> GetActivities()
+        {
+            var activities = _context.Activity
+                            .Select(a => new ActivityExtendedViewModel
+                            {
+                                ID = a.ID,
+                                Name = a.Name,
+                                RegistrationStart = a.RegistrationStart,
+                                RegistrationEnd = a.RegistrationEnd,
+                                RegistrationOpen = DateTime.Now > a.RegistrationStart && DateTime.Now < a.RegistrationEnd,
+                                Sport = _context.Sport
+                                        .FirstOrDefault(s => s.ID == a.SportID),
+                                Status = _context.ActivityStatus
+                                        .FirstOrDefault(s => s.ID == a.StatusID)
+                                        .Description,
+                                MinCapacity = a.MinCapacity,
+                                MaxCapacity = a.MaxCapacity,
+                                SoloRegistration = a.SoloRegistration,
+                                Logo = a.Logo,
+                                Completed = a.Completed,
+                                Series = a.Series
+                                        .Select(s => new SeriesExtendedViewModel
+                                        {
+                                            ID = s.ID,
+                                            Name = s.Name,
+                                            StartDate = s.StartDate,
+                                            EndDate = s.EndDate,
+                                            Type = _context.SeriesType
+                                                    .FirstOrDefault(st => st.ID == s.TypeID)
+                                                    .Description
+                                        }).ToList(),
```

Comment:
I was modifying this before i saw this comment. Is there a stark difference if I do an explicit cast versus calling the constructor?

i.e.

`(SeriesExtendedViewModel)s`
vs
`new SeriesExtendedViewModel(s)`

Reply:
There is a significant difference, because the former tries to perform type conversion and throws an exception if that fails, whereas the latter calls the Constructor. So it's a question of whether you want to explicitly call the Constructor method, or implicitly cast, hoping that there is a conversion operator (e.g. an `implicit operator SeriesExtendViewModel(Series s)`) defined that will not error.

Reply:
I see, the cases we've had where we do a `Select(_ -> (viewmodel)_).ToList` have always had implicit operators. But I'll keep what you said in mind. I honestly didn't know you could just call a default constructor to cast it for you. (Unless I'm reading this wrong)

Reply:
Yeah, we've used implicit operators a lot in the API. Either way is idiomatic, so I don't particularly care.

> call a default constructor to cast it for you
This is not exactly right. You would need to write the constructor, there isn't a default constructor from another type. So either way, you need to write about the same amount of code. It's mostly a question of whether you want to write the conversion explicitly, or tell the compiler to cast, assuming that at some level there is a conversion operator implemented to handle that.